types/shares_test.go

```diff
@@ -439,6 +438,26 @@ func Test_parseMsgShares(t *testing.T) {
 	}
 }
 
+func TestParsePaddedMsg(t *testing.T) {
+	msgWr := NewMessageShareWriter()
+	randomSmallMsg := generateRandomMessage(100)
+	randomLargeMsg := generateRandomMessage(10000)
+	msgs := Messages{
+		MessagesList: []Message{
+			randomSmallMsg,
+			randomLargeMsg,
+		},
+	}
+	msgs.SortMessages()
+	msgWr.Write(msgs.MessagesList[0])
+	msgWr.WriteNamespacedPaddedShares(4)
+	msgWr.Write(msgs.MessagesList[1])
+	msgWr.WriteNamespacedPaddedShares(10)
```

Comment:
[question for my understanding] would the previous implementation fail if the message share writer contained these namespaced padded shares? I recognize the entire test is new so we don't have a failing run with the previous `parseMsgShares` implementation

Reply:
yeah, the previous implementation assumes that if we find a share with message with a delimiter length of zero (messages are [prepended with a length delimiter](https://github.com/celestiaorg/celestia-core/blob/481fd21db7dfeb2d4312692ead148ccedd2951e7/types/block.go#L1168)) then the rest of the message are tailpadding.